src/lib/Components/Bidding/Components/Timer.tsx

```diff
@@ -18,10 +21,23 @@ export class Timer extends React.Component<TimerProps, TimerState> {
   constructor(props) {
     super(props)
 
-    const { liveStartsAt, endsAt } = props
-    const timeLeftInMilliseconds = Date.parse(liveStartsAt || endsAt) - Date.now()
+    const { liveStartsAt, endsAt, isPreview, startsAt } = props
 
-    this.state = { timeLeftInMilliseconds }
+    let relevantDate
+    let upcomingLabel
+
+    if (isPreview) {
+      relevantDate = startsAt
+      upcomingLabel = `Starts ${this.formatDate(startsAt)}`
```

Comment:
It may be helpful to `console.error` when a required prop is missing. I know it's not clean but when a flag is passed in it is always tricky to confirm the interface properly:

```ts
if (isPreview) {
  if (!startsAt) {
    console.error("startsAt is required when isPreview is true")
  }

  relevantDate = startsAt
  upcomingLabel = `Starts ${this.formatDate(startsAt)}`
} else if (liveStartsAt) {
  relevantDate = liveStartsAt
  upcomingLabel = `Live ${this.formatDate(liveStartsAt)}`
} else if (endsAt) {
  relevantDate = endsAt
  upcomingLabel = `Ends ${this.formatDate(endsAt)}`
} else {
  console.error("liveStartsAt or endsAt is required when isPreview is false")
}
```